### PR TITLE
Fix: Activate zoom out on large viewport

### PR DESCRIPTION
**Pull request by PARTHVATALIYA**
## What?
Solve https://github.com/WordPress/gutenberg/issues/66205

## Why?
Part of https://github.com/WordPress/gutenberg/issues/66205

## Testing Instructions
- Open the post editor.
- Open the main inserter.
- Click the Patterns tab.

### CODE REVIEW

**Review by getdave — COMMENTED** [review]
Thank you for this PR 👍 

To test this accurately I needed to set my viewport to around `587px`. Is that right?

Here's what I saw

## Before (trunk)
Screenshot 2024-10-22 at 10 08 43

## After (this branch)
Screenshot 2024-10-22 at 10 09 42

Once this is merged we'll need a similar fix targeting `wp/6.7` branch. Let's wait for this one to merge before we tackle that though so as not to duplicate effort.

Thanks again for your contribution here 🙇

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR! 

I think we need to add `isWideViewport` to the useEffect hook dependencies ([here](https://github.com/WordPress/gutenberg/blob/0e65adcbe187797a536aa99f56b063a61cafdc3d/packages/block-editor/src/hooks/use-zoom-out.js#L34) and [here](https://github.com/WordPress/gutenberg/blob/0e65adcbe187797a536aa99f56b063a61cafdc3d/packages/block-editor/src/hooks/use-zoom-out.js#L42)) to completely solve this issue, otherwise the inserter width will not change if we change the viewport width with the Patterns tab open:

https://github.com/user-attachments/assets/4b718b53-b4d2-4585-bd03-2c7f4f6a94ab

File: packages/block-editor/src/hooks/use-zoom-out.js
Comment (getdave) [inline packages/block-editor/src/hooks/use-zoom-out.js:36]:
```suggestion
	}, [ isWideViewport ] );
```

Comment (t-hamano) [reply packages/block-editor/src/hooks/use-zoom-out.js:36]: This useEffect looks like it's only needed on initial render. Maybe we could just add the dependency [here](https://github.com/WordPress/gutenberg/pull/66308/files#diff-c22c80cca155dff13e53f4a897ba8671d92bcf00eed692405ff391dbf5211673R44) instead?

Comment (PARTHVATALIYA) [reply packages/block-editor/src/hooks/use-zoom-out.js:36]: I’ve just added the dependency [here](https://github.com/WordPress/gutenberg/pull/66308/files#diff-c22c80cca155dff13e53f4a897ba8671d92bcf00eed692405ff391dbf5211673R44). Now, whenever the pattern tab is open and the viewport changes from large to medium, the zoom-out is reset.

**Review by t-hamano — APPROVED** [review]
LGTM!

@getdave Does this PR need a backport label? Is it okay to merge it as is, since the automatic cherry-pick will probably fail?

Either way, I think we need a separate PR for `wp/6.7`.

**Comment by t-hamano** [comment]
Merged trunk to fix lint errors. See #66316.

**Comment by t-hamano** [comment]
Just to be safe, I will remove the backport label before merging. I will create a PR for WP 6.7 now.

**Comment by getdave** [comment]
@t-hamano Yes let's do a manual backport and then add to the 6.7 board for review. Thanks

**Comment by getdave** [comment]
Manual backport in https://github.com/WordPress/gutenberg/pull/66330

**Comment by jeryj** [comment]
I don't think this is the fix we want. I think the viewport check should be passed into the `useZoomOut()` hook. I opened a PR to demonstrate what I mean: https://github.com/WordPress/gutenberg/pull/66341

**Pull request merged**
